bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GenericItem.java

```diff
@@ -401,15 +403,44 @@ public StateDescription getStateDescription() {
 
     @Override
     public StateDescription getStateDescription(Locale locale) {
+        StateDescription fullStateDescription = null;
         if (stateDescriptionProviders != null) {
+            BigDecimal minimum = null;
+            BigDecimal maximum = null;
+            BigDecimal step = null;
+            String pattern = null;
+            boolean readOnly = false;
+            List<StateOption> options = null;
+            boolean found = false;
             for (StateDescriptionProvider stateDescriptionProvider : stateDescriptionProviders) {
                 StateDescription stateDescription = stateDescriptionProvider.getStateDescription(this.name, locale);
                 if (stateDescription != null) {
-                    return stateDescription;
+                    found = true;
+                    if (minimum == null) {
+                        minimum = stateDescription.getMinimum();
+                    }
+                    if (maximum == null) {
+                        maximum = stateDescription.getMaximum();
+                    }
+                    if (step == null) {
+                        step = stateDescription.getStep();
+                    }
+                    if (pattern == null) {
+                        pattern = stateDescription.getPattern();
+                    }
+                    if (stateDescription.isReadOnly()) {
+                        readOnly = true;
+                    }
+                    if (options == null) {
+                        options = stateDescription.getOptions();
+                    }
                 }
             }
+            if (found) {
+                fullStateDescription = new StateDescription(minimum, maximum, step, pattern, readOnly, options);
```

Comment:
The idea here is to mix up state descriptions of different providers, right? 
This will lead to quite unexpected results, therefore I don't think this is a good idea. Also I wouldn't want to mix this into a PR which intends to add a default state description at all (or I did not understand how it is related).